common/app/common/configuration.scala

```diff
@@ -1,33 +1,71 @@
 package common
 
 import java.io.{File, FileInputStream}
+import java.util.Map.Entry
 
 import com.amazonaws.AmazonClientException
 import com.amazonaws.auth._
 import com.amazonaws.auth.profile.ProfileCredentialsProvider
-import com.gu.conf.ConfigurationFactory
+import com.gu.cm.{PlayDefaultLogger, ClassPathConfigurationSource, FileConfigurationSource, Logger}
+import com.typesafe.config.ConfigException
 import conf.switches.Switches
 import conf.{Configuration, Static}
 import org.apache.commons.io.IOUtils
 import play.api.Play
 
-import scala.util.Try
+import scala.collection.JavaConversions._
+import scala.util.{Failure, Success, Try}
 
 class BadConfigurationException(msg: String) extends RuntimeException(msg)
 
-class GuardianConfiguration extends Logging {
-  implicit private lazy val app = Play.current
+object GuardianConfiguration extends Logging {
 
-  case class OAuthCredentials(oauthClientId: String, oauthSecret: String, oauthCallback: String)
-  case class OAuthCredentialsWithMultipleCallbacks(oauthClientId: String, oauthSecret: String, authorizedOauthCallbacks: List[String])
+  import com.gu.cm.{Configuration => CM}
+  import com.typesafe.config.Config
+  lazy val configuration = {
+
+    val stage = new CM(List(FileConfigurationSource(s"/etc/gu/install_vars")), PlayDefaultLogger).load.getStringProperty("STAGE").getOrElse("DEV")
+    lazy val userPrivate = FileConfigurationSource(s"${System.getProperty("user.home")}/.gu/frontend.properties")
+    lazy val opsPrivate = FileConfigurationSource(s"/etc/gu/frontend.properties")
+    lazy val public = ClassPathConfigurationSource(s"env/$stage.properties")
+    new CM(List(userPrivate, opsPrivate, public), PlayDefaultLogger).load
+  }
+
+  implicit class ScalaConvertProperties(conf: Config) {
```

Comment:
I think you don't need any of these if instead of exposing a `Config` you expose a `play.api.Configuration` by doing `playConf = new Configuration(config)`

That should give you all the scala helpers you need.